### ISSUE TITLE
`doc_markdown` should accept `<code>` as equivalent to backticks

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust-clippy#9473]
### Summary

If one wishes to include links to a type with parameters, or within a code snippet, then to ensure that the code fragment is spanned by one entire `<code>` element (which is important particularly when stylesheets add whitespace or borders around `<code>` elements), one must deviate from the typical Markdown pattern of <code>\[\`SomeIdentifier\`\]</code> and use explicit HTML (that Markdown links can live within). But the `clippy::doc_markdown` lint thinks that this is an identifier without code formatting.

Following clippy's suggestion would result in the final HTML containing a nonsensical doubly nested `<code>` element.

Instead, it should recognize `<code>` as an adequate substitute for backticks.

### Lint Name

doc_markdown

### Reproducer

I tried this code:

```rust
#![warn(clippy::doc_markdown)]
use std::collections::HashSet;
use std::sync::atomic::AtomicUsize;

/// Blah blah blah <code>[HashSet]&lt;[AtomicUsize]&gt;</code>.
pub struct Foo(HashSet<AtomicUsize>);
```

I saw this happen:

```
warning: item in documentation is missing backticks
 --> src/lib.rs:5:27
  |
5 | /// Blah blah blah <code>[HashSet]&lt;[AtomicUsize]&gt;</code>.
  |                           ^^^^^^^
  |
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(clippy::doc_markdown)]
  |         ^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown
help: try
  |
5 | /// Blah blah blah <code>[`HashSet`]&lt;[AtomicUsize]&gt;</code>.
  |                           ~~~~~~~~~

warning: item in documentation is missing backticks
 --> src/lib.rs:5:40
  |
5 | /// Blah blah blah <code>[HashSet]&lt;[AtomicUsize]&gt;</code>.
  |                                        ^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown
help: try
  |
5 | /// Blah blah blah <code>[HashSet]&lt;[`AtomicUsize`]&gt;</code>.
  |                                        ~~~~~~~~~~~~~
```

Neither of these warnings should be emitted.

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-apple-darwin
release: 1.63.0
LLVM version: 14.0.5

Also reproduces on nightly.
```
